### PR TITLE
feature: add userpilot to TAO 3.x, save tenant id as session context

### PR DESCRIPTION
**Pull request by pribi**
Ticket: https://oat-sa.atlassian.net/browse/MS-2756

## What's Changed
- Added UserPilot to TAO 3.x.

## TODO 

- [x] Unit tests
- [x] E2E tests
- [x] Update composer with packages

## Dependencies PRs
- https://github.com/oat-sa/tao-core/pull/3967

## How to test
- Install TAO Portal and TAO 3.x Authoring using NGS.
- cd nextgen-stack/tao
- composer require "oat-sa/extension-tao-lti":"dev-feature/MS-2756/add-userpilot-to-tao-3.x as 15.15.1" "oat-sa/tao-core":"dev-feature/MS-2756/add-userpilot-to-tao-3.x as 54.4.1"
- Setup TAO Portal - Authoring simplification
- Login to TAO Portal
- Go to Authoring by clicking on Content Bank
- Inspect source code

## Screenshots
image

### CODE REVIEW

**Comment by viktar-dzmitryieu-tao** [comment]
should it point to `develop` branch instead?

File: composer.json
Comment (andreluizmachado) [inline composer.json:64]: Do not forget to replace by the proper release before merge

**Review by andreluizmachado — CHANGES_REQUESTED** [review]
Great job, please just do not forget to change the target branch. It should be develop, not master

File: composer.json
Comment (pribi) [reply composer.json:64]: Yes, I am aware of it. First I need approval for depending PR, merge it, and then update the new version here.

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: composer.json
Comment (pribi) [reply composer.json]: Updated to >=54.8.0

**Pull request merged**
